java/src/main/native/include/jni_utils.hpp

```diff
@@ -790,19 +756,42 @@ inline void jni_cuda_check(JNIEnv *const env, cudaError_t cuda_status) {
+    }                                                                                              \
+    std::string n_msg = std::string("CUDA ERROR: ") + (e.what() == nullptr ? "" : e.what());       \
+    jstring j_msg = env->NewStringUTF(n_msg.c_str());                                              \
+    const char *n_name = cudaGetErrorName(e.error_code());                                         \
+    jstring j_name = env->NewStringUTF(n_name);                                                    \
+    jclass ex_class = env->FindClass(class_name);                                                  \
+    jmethodID ctor_id =                                                                            \
+        env->GetMethodID(ex_class, "<init>", "(Ljava/lang/String;Ljava/lang/String;)V");           \
+    if (ctor_id == NULL) {                                                                         \
+      return ret_val;                                                                              \
+    }                                                                                              \
+    jobject cuda_error = env->NewObject(ex_class, ctor_id, j_msg, j_name);                         \
+    env->Throw((jthrowable)cuda_error);                                                            \
+    return ret_val;                                                                                \
+  } while (0)
+
 #define JNI_CUDA_TRY(env, ret_val, call)                                                           \
-  {                                                                                                \
+  do {                                                                                             \
     cudaError_t internal_cuda_status = (call);                                                     \
     if (cudaSuccess != internal_cuda_status) {                                                     \
-      /* Clear the last error so it does not propagate.*/                                          \
-      cudaGetLastError();                                                                          \
-      jthrowable jt = cudf::jni::cuda_exception(env, internal_cuda_status);                        \
-      if (jt != NULL) {                                                                            \
-        env->Throw(jt);                                                                            \
-      }                                                                                            \
+      cudf::detail::throw_cuda_error(internal_cuda_status, __FILE__, __LINE__);                    \
       return ret_val;                                                                              \
     }                                                                                              \
-  }
+  } while (0)
+
+#define JNI_CUDA_CHECK(env, cuda_status)                                                           \
```

Comment:
This macro takes an `env` argument that is never used.  Was the intent to have this macro throw a JVM exception?  If so it needs to be used, otherwise if it's just going to throw a native exception the arg should be removed.

Reply:
Redundant `env`s are removed.